src/mbgl/util/mapbox.cpp

```diff
@@ -39,6 +39,20 @@ std::string normalizeGlyphsURL(const std::string& url, const std::string& access
     return normalizeURL(url, accessToken);
 }
 
+std::string normalizeTileURL(const std::string& url) {
+    if (url.compare(url.length() - 4, 4, ".png") == 0)
```

Comment:
It's not just `.png` that's relevant here. 

https://www.mapbox.com/developers/api/maps/#format